src/groupeddataframe/grouping.jl

```diff
@@ -1106,3 +1112,19 @@ function _aggregate(d::AbstractDataFrame, fs::AbstractVector,
     sort && sort!(res, headers)
     res
 end
+
+function DataFrame(gd::GroupedDataFrame)
+    length(gd) == 0 && return similar(parent(gd), 0)
+    idx = similar(gd.idx)
+    doff = 1
+    min_start = gd.starts[1]
+    for (s,e) in zip(gd.starts, gd.ends)
+        n = e - s + 1
+        copyto!(idx, doff, gd.idx, s, n)
+        doff += n
+        min_start = min(min_start, s)
+    end
+    resize!(idx, doff - 1)
+    @assert doff == length(gd.idx) + 2 - min_start
```

Comment:
It's kind of weird to have this assertion here. Why not put something equivalent in `groupby_checked` instead? That way we will check it much more systematically, and the code here will be simpler.

Reply:
An excellent point. I have included a combo of tests to `groupby_checked` that I think are also useful for the future readers to understand what we expect to have in `GroupedDataFrame` object.